lib/_http_incoming.js

```diff
@@ -315,6 +315,8 @@ IncomingMessage.prototype._dump = function _dump() {
   if (!this._dumped) {
     this._dumped = true;
     this.resume();
+    // Flush the existing buffered data
+    this._readableState._flush();
```

Comment:
@mscdex I do not want to rush adding a new method to an already complex API to fix a bug either. I'm ok in adding this, just not as a bug fix, as it will need to be reasoned upon for a bit. If this can wait, I'm ok following that path.